runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -741,6 +741,7 @@ J9::Z::TreeEvaluator::inlineVectorizedStringIndexOf(TR::Node* node, TR::CodeGene
       cursor = generateRIInstruction(cg, TR::InstOpCode::getAddHalfWordImmOpCode(), node, loadLenReg, -1);
       iComment("needs -1 because VLL's third operand is the highest index to load");
       generateVRSbInstruction(cg, TR::InstOpCode::VLL, node, stringVReg, loadLenReg, generateS390MemoryReference(stringCharPtrReg, headerSize, cg));
+      generateRIInstruction(cg, TR::InstOpCode::getAddHalfWordImmOpCode(), node, loadLenReg, 1);
```

Comment:
The reason this change is needed is because we have two paths from `labelFindPatternHead`.

1. There are < 16 characters left to load from search string
2. There are >= 16 characters left to load from search string

In (2) the value of `loadLenReg` is always 16, and is the number of characters we loaded. In (1) however the `loadLenReg` register contains one less than the number of characters we ended up loading because of the way `VLL` instruction works. We must compensate for this by incrementing this register after the load so that the logic after `VSTRS` is consistent in that  `loadLenReg` contains the number of characters we just loaded.

Reply:
Agreed with the need to add 1 to re-adjust `loadLenReg` for case (1).  Thanks for the added explanation.